subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -17,37 +17,145 @@
+#           manner to JSP or MASON or PHP: anything within < % ... % > is a
+#           code block, anything else is sent as output, subject to the
+#           flow imposed by any code segments.
+#
+#           Yes, if you need a \ in the second phase, you need to write \\
+#
+#       See https://docs.groovy-lang.org/next/html/documentation/ section §3.15
+#       (Template Engines) for details.
+#
+#       Bear in mind that the "output" is a POSIX shell script, which has its
+#       own ideas about escaping with backslashes, so to get «\» you need to
+#       write «\\\\» and to get «$» you need to write «\\\$».
+#
+#       Code segments are used in this script to control conditional inclusions:
+#           (a) to build a module rather than an app, indicated by passing
+#               '--module XYZ' in place of the classname that contains main
+#           (b) to optionally define a ${appNameSystemProperty} symbol
+#
+#       These are the variables used in this script, and the defaults when
+#       building Gradle itself:
+#
+#           ${appNameSystemProperty}    org.gradle.appname
+#           ${applicationName}          Gradle
+#           ${mainClassName}            org.gradle.wrapper.GradleWrapperMain
+#           ${optsEnvironmentVar}       GRADLE_OPTS
+#           ${defaultJvmOpts}           -Dfile.encoding=UTF-8 "-Xmx64m" "-Xms64m"
+#
+#       (These are all specified in
+#       subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java
+#       within the Gradle project; they may be different for other projects.)
+# */ %>
+#       You can find Gradle at https://github.com/gradle/gradle/.
+#
+#   (3) This script targets any POSIX shell, not just Bash or Ksh.
+#
+#       It makes use of these POSIX shell features:
+#         * functions;
+#         * various expansions, including «$var», «$( cmd )», «${var#prefix}»,
+#           and «${var%suffix}»;
+#         * compound commands having a testable exit status, especially «case»;
+#         * various built-in commands including «command», «set», and «ulimit».
+#
+#       However arrays are NOT available, except for "\$@".
+#
+#       The traditional practice of packing multiple parameters into a
+#       space-separated string is a well documented source of bugs and security
+#       problems, so this is avoided.
+#
+#       Instead, this script progressively modifies "\$@", which is eventually
+#       passed to Java.
+#
+#       The is special handling for the inherited environment variables
+#       DEFAULT_JVM_OPTS, JAVA_OPTS, and ${optsEnvironmentVar}; see the in-line
+#       comments for details.
+#
+#       There are tweaks for specific operating systems such as AIX, CygWin,
+#       Darwin, MinGW, and NonStop.
+#
+#       If submitting patches, please make sure that this script (as output by
+#       the template generator) still passes shellcheck
+#       (https://shellcheck.net)
```

Comment:
I think we can enforce this on our CI infrastructure as well now that the script passes. You don't need to figure that out.